Sources/NostrSDK/Helpers/String+Additions.swift

```diff
@@ -31,4 +31,47 @@ public extension String {
 
         return data
     }
+
+    /// Returns ``DateComponents`` from a string representation of a date in the format of yyyy-mm-dd
+    /// or nil if the string does not match the format.
+    var dateStringAsDateComponents: DateComponents? {
```

Comment:
I would suggest that this instead be a failable initializer in an extension on `DateComponents`.

Would start like this:
```swift
extension DateComponents {
    convenience init?(dateString: String) { ... }
}
```

You can return nil inside that. Then you'd `guard let` or `if let` when you use the initializer:
```swift
if let comps = DateComponents(dateString: <val>) { ... }
```

Reply:
Oh, that's a way better pattern.

Reply:
Done! `convenience` doesn't apply to structs, only classes, so I left that out.